src/actions/receive-profile.js

```diff
@@ -1620,6 +1647,31 @@ export function retrieveProfileForRawUrl(
         }
         break;
       }
+      case 'from-post-message': {
+        window.addEventListener('message', (event) => {
+          const { data } = event;
+          console.log(`Received postMessage`, data);
+          if (!data || typeof data !== 'object') {
+            return;
+          }
+          switch (data.name) {
+            case 'inject-profile':
+              dispatch(viewProfileFromPostMessage(data.profile));
+              break;
+            case 'is-ready': {
+              console.log(
+                'Responding via postMessage that the profiler is ready.'
+              );
+              const otherWindow = event.source ?? window;
+              otherWindow.postMessage({ name: 'is-ready' }, '*');
+              break;
+            }
```

Comment:
~~See my message above: instead of the block for `is-ready`, we could do something like~~
```js
if (!window.opener) {
  throw new Error("No window.opener is available in this window");
}
window.opener.postMessage({ name: 'is-ready' }, '*');
```
~~right after adding the event handler.~~

~~This would also provide some diagnostic to the user in case something doesn't work as planned.~~

**Update**: now I understand this is needed for addons and iframes. Can you please add a comment precising these various use cases?